mobile/android/android-device.ts

```diff
@@ -185,9 +188,166 @@ export class AndroidDevice implements Mobile.IDevice {
+            if (match) {
+                port = match[0].substring(7);
+            } else {
+                port = 0;
+            }
+        }
+        if (port > 0) {
+            this.tcpForward(port, port).wait();
+        }
+        this.startDebuggerClient(port).wait();
+        this.openDebuggerClient(this.defaultNodeInspectorUrl + "?port=" + port).wait();
+    }
+
+    private detachDebugger(packageName: string): void {
+        var stopDebuggerCommand = this.composeCommand("shell am broadcast -a \"%s-Debug\" --ez enable false", packageName);
+        this.$childProcess.exec(stopDebuggerCommand).wait();
+    }
+
+    private startAppWithDebugger(packageFile: string, packageName: string): void {
+        var uninstallCommand = this.composeCommand("shell pm uninstall \"%s\"", packageName)
+			    this.$childProcess.exec(uninstallCommand).wait();
+
+        var installCommand = this.composeCommand("install -r \"%s\"", packageFile);
+        this.$childProcess.exec(installCommand).wait();
+
+        var port = options["debug-port"];
+
+        var clearDebugEnvironmentCommand = this.composeCommand('shell rm "/sdcard/Android/data/%s/files/envDebug.out"', packageName);
+        this.$childProcess.exec(clearDebugEnvironmentCommand).wait();
+
+        this.startPackageOnDevice(packageName).wait();
+
+        var dbgPort = this.startAndGetPort(packageName).wait();
+
+        if (dbgPort > 0) {
+            this.tcpForward(dbgPort, dbgPort).wait();
+            this.startDebuggerClient(dbgPort).wait();
+            this.openDebuggerClient(this.defaultNodeInspectorUrl + "?port=" + dbgPort).wait();
+        }
+    }
+
+    public debug(packageFile: string, packageName: string): IFuture<void> {
+        return (() => {
+            if (options["get-port"]) {
+                this.printDebugPort(packageName);
+            } else if (options["start"]) {
+                this.attachDebugger(packageName);
+            } else if (options["stop"]) {
+                this.detachDebugger(packageName);
+            } else if (options["debug-brk"]) {
+                this.startAppWithDebugger(packageFile, packageName);
+            } else {
+                this.$logger.info("Should specify at least one option: debug-brk, start, stop, get-port.");
+            }
+        }).future<void>()();
+    }
+
+    private checkIfRunning(packageName: string): boolean {
+        var isRunning = this.checkIfFileExists(packageName, "envDebug.out").wait();
+        return isRunning;
+    }
+
+    private readDebugPort(packageName: string): IFuture<number> {
+        return (() => {
+            var res = this.$childProcess.spawnFromEvent(this.adb, ["shell", "cat", "/sdcard/Android/data/" + packageName + "/files/envDebug.out"], "exit").wait();
+            var isRunning = res.stdout.indexOf('yes') > -1;
+            return 1;
+        }).future<number>()();
+    }
+
+    private sleep(ms: number): void {
+        var fiber = Fiber.current;
```

Comment:
Done with https://github.com/telerik/mobile-cli-lib/commit/1a8d23f43fcd26503d703143ff750445c94f88a8#diff-b823ab15d1844ca2c6b85896ef654162R201 and https://github.com/telerik/mobile-cli-lib/commit/1a8d23f43fcd26503d703143ff750445c94f88a8#diff-63f4603786adf475e860b557e6e5ea98R316